### ISSUE TITLE
could not run models on a offline server successfully

### ISSUE THREAD

**Issue by lkcao** [huggingface/datasets#1724]
Hi, I really need your help about this.
I am trying to fine-tuning a RoBERTa on a remote server, which is strictly banning internet. I try to install all the packages by hand and try to run run_mlm.py on the server. It works well on colab, but when I try to run it on this offline server, it shows:
![image](https://user-images.githubusercontent.com/49967236/104276256-25a88600-546a-11eb-9776-8ec695dfa24e.png)

is there anything I can do? Is it possible to download all the things in cache and upload it to the server? Please help me out...

**Comment by julien-c** [comment]
Transferred to `datasets` based on the stack trace.

**Comment by SBrandeis** [comment]
Hi @lkcao !
Your issue is indeed related to `datasets`. In addition to installing the package manually, you will need to download the `text.py` script on your server. You'll find it (under `datasets/datasets/text`: https://github.com/huggingface/datasets/blob/master/datasets/text/text.py.
Then you can change the line 221 of `run_mlm_new.py` into:
```python
  datasets = load_dataset('/path/to/text.py`, data_files=data_files)
```
Where `/path/to/text.py` is the path on the server where you saved the `text.py` script.

**Comment by lhoestq** [comment]
We're working on including the local dataset builders (csv, text, json etc.) directly in the `datasets` package so that they can be used offline

**Comment by HughWen** [comment]
> Hi @lkcao !
> Your issue is indeed related to `datasets`. In addition to installing the package manually, you will need to download the `text.py` script on your server. You'll find it (under `datasets/datasets/text`: https://github.com/huggingface/datasets/blob/master/datasets/text/text.py.
> Then you can change the line 221 of `run_mlm_new.py` into:
> 
> ```python
>   datasets = load_dataset('/path/to/text.py', data_files=data_files)
> ```
> 
> Where `/path/to/text.py` is the path on the server where you saved the `text.py` script.

Hi SBrandeis, when I do this for 'csv.py' such as 
`dataset = load_dataset('./csv.py', data_files=file_path, split='train')`
I get the error like 
AttributeError: module 'datasets' has no attribute 'BuilderConfig'

**Comment by xinjicong** [comment]
> The local dataset builders (csv, text , json and pandas) are now part of the `datasets` package since #1726 :)
> You can now use them offline
> 
> ```python
> datasets = load_dataset('text', data_files=data_files)
> ```
> 
> We'll do a new release soon

so the new version release now?

**Comment by xinjicong** [comment]
> The local dataset builders (csv, text , json and pandas) are now part of the `datasets` package since #1726 :)
> You can now use them offline
> 
> ```python
> datasets = load_dataset('text', data_files=data_files)
> ```
> 
> We'll do a new release soon

so the new version release now?

**Comment by lhoestq** [comment]
Yes it's been available since datasets 1.3.0 !
